game-core/src/main/java/games/strategy/engine/data/unit/ability/ConvertUnitAbility.java

```diff
@@ -0,0 +1,64 @@
+package games.strategy.engine.data.unit.ability;
+
+import games.strategy.engine.data.UnitType;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.NonFinal;
+
+/** Unit Ability to convert CombatUnitAbilities into other CombatUnitAbilities or into nothing */
+@Data
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Builder(toBuilder = true)
+public class ConvertUnitAbility {
+
+  enum Faction {
```

Comment:
'Faction' could be overloaded or confused. Consider renaming. 'Faction' is the leading contender for renaming what is known as a  'Player'

Reply:
The support attachments use `faction` so I copied that term from there.  I couldn't think of another term to describe.  So if you have suggestions, please give them.

Reply:
Side and team are best i can see from a thesaurus: https://www.thesaurus.com/browse/faction

I agree `faction` fits, though I would like that reserved for the time being as the replacement for `player`. Once that happens, it would be confusing to see it used in a different way.

Perhaps some ideas may come to us if we sleep on it.

Reply:
Relationship might work.  Side already has a meaning of "offense" and "defense" so I wouldn't want to use it here.

Reply:
Affiliation is another possible word.

Reply:
Couple more suggestions:
`team { friendly | foe }`
`FriendOrFoe { friendly | foe }`

Reply:
I renamed it to Team - https://github.com/triplea-game/triplea/pull/8646/commits/a75da36aa7e742dd907b21ad0eef9d98e6a3ebc8